util/src/main/kotlin/DiskCache.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2017 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.util
+
+import ch.frankel.slf4k.error
+
+import com.jakewharton.disklrucache.DiskLruCache
+
+import java.io.File
+import java.io.IOException
+
+/**
+ * Wrapper around [DiskLruCache] that adds a workaround for the 64 character key length limit.
+ */
+class DiskCache(
+        /**
+         * The directory to store the cache data, must be exclusive to the cache.
+         */
+        val directory: File,
+
+        /**
+         * The maximum size of the disk cache.
+         */
+        val maxSize: Long,
+
+        /**
+         * Duration in seconds that cache entries are valid.
+         */
+        val timeToLive: Int
+) {
+    companion object {
+        const val INDEX_FULL_KEY = 0
+        const val INDEX_TIMESTAMP = 1
+        const val INDEX_DATA = 2
+        const val VALUE_COUNT = 3
+    }
+
+    private val diskLruCache = DiskLruCache.open(directory, 0, VALUE_COUNT, maxSize)
+
+    /**
+     * Shorten the string to be usable as a key for [DiskLruCache] which has a maximum length of 64. If the string is
```

Comment:
Looks like the limit is 120 now: https://github.com/JakeWharton/DiskLruCache/commit/057db125d55825d81ec17f58b85e976ee38df438#diff-04c6e90faac2675aa89e2176d2eec7d8

Reply:
Unfortunately, there's no official release with that change. We could use JitPack to dynamically create one, or fork the project an publish our own release.

Reply:
We would still need to implement the same logic, as keys can also be longer than 120 chars, so I think we wouldn't win much.

There is also https://github.com/JakeWharton/DiskLruCache/issues/98, but it seems Jake is not active on this project anymore.